From 65a851f41c8627881d9d4085eea43806e2af7e68 Mon Sep 17 00:00:00 2001
From: Mikhail Ulyanov <mikhail.ulyanov@cogentembedded.com>
Date: Tue, 19 Aug 2014 16:50:49 +0400
Subject: [PATCH 22/25] ARM: shmobile: r8a7790: Add JPU clock dt and CPG
 define.

Signed-off-by: Mikhail Ulyanov <mikhail.ulyanov@cogentembedded.com>
Acked-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
(cherry picked from commit da076a888ab19f13816372796ed231e7d6ff5fed)
Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
Signed-off-by: Andrey Gusakov <andrey.gusakov@cogentembedded.com>
---
 arch/arm/boot/dts/r8a7790.dtsi            |    6 +++---
 include/dt-bindings/clock/r8a7790-clock.h |    1 +
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/r8a7790.dtsi b/arch/arm/boot/dts/r8a7790.dtsi
index c0675d8..91fb4a7 100644
--- a/arch/arm/boot/dts/r8a7790.dtsi
+++ b/arch/arm/boot/dts/r8a7790.dtsi
@@ -875,14 +875,14 @@
 			compatible = "renesas,r8a7790-mstp-clocks", "renesas,cpg-mstp-clocks";
 			reg = <0 0xe6150134 0 4>, <0 0xe6150038 0 4>;
 			clocks = <&zs_clk>, <&zs_clk>, <&zs_clk>, <&zs_clk>,
-				 <&zs_clk>, <&p_clk>, <&zg_clk>, <&zs_clk>,
+				 <&m2_clk>, <&zs_clk>, <&p_clk>, <&zg_clk>, <&zs_clk>,
 				 <&zs_clk>, <&zs_clk>, <&zs_clk>, <&p_clk>,
 				 <&p_clk>, <&rclk_clk>, <&cp_clk>, <&zs_clk>,
 				 <&zs_clk>, <&zs_clk>, <&zs_clk>;
 			#clock-cells = <1>;
 			renesas,clock-indices = <
 				R8A7790_CLK_VCP1 R8A7790_CLK_VCP0 R8A7790_CLK_VPC1
-				R8A7790_CLK_VPC0 R8A7790_CLK_SSP1 R8A7790_CLK_TMU1
+				R8A7790_CLK_VPC0 R8A7790_CLK_JPU R8A7790_CLK_SSP1 R8A7790_CLK_TMU1
 				R8A7790_CLK_PVRSRVKM R8A7790_CLK_2DDMAC R8A7790_CLK_FDP2
 				R8A7790_CLK_FDP1 R8A7790_CLK_FDP0 R8A7790_CLK_TMU3
 				R8A7790_CLK_TMU2 R8A7790_CLK_CMT0 R8A7790_CLK_TMU0
@@ -890,7 +890,7 @@
 				R8A7790_CLK_VSP1_S
 			>;
 			clock-output-names =
-				"vcp1", "vcp0", "vpc1", "vpc0", "ssp_dev",
+				"vcp1", "vcp0", "vpc1", "vpc0", "jpu", "ssp_dev",
 				"tmu1", "pvrsrvkm", "tddmac", "fdp2", "fdp1",
 				"fdp0", "tmu3", "tmu2", "cmt0", "tmu0",
 				"vsp1-du1", "vsp1-du0", "vspr", "vsps";
diff --git a/include/dt-bindings/clock/r8a7790-clock.h b/include/dt-bindings/clock/r8a7790-clock.h
index 42cc15b..f7d0639 100644
--- a/include/dt-bindings/clock/r8a7790-clock.h
+++ b/include/dt-bindings/clock/r8a7790-clock.h
@@ -30,6 +30,7 @@
 #define R8A7790_CLK_VCP0		1
 #define R8A7790_CLK_VPC1		2
 #define R8A7790_CLK_VPC0		3
+#define R8A7790_CLK_JPU			6
 #define R8A7790_CLK_SSP1		9
 #define R8A7790_CLK_TMU1		11
 #define R8A7790_CLK_PVRSRVKM		12
-- 
1.7.10.4

